cpp/src/search/search.cu

```diff
@@ -0,0 +1,239 @@
+  auto d_values = table_device_view::create(values, stream);
+  auto count_it = thrust::make_counting_iterator<size_type>(0);
+
+  rmm::device_vector<order> d_column_order(column_order.begin(), column_order.end());
+  rmm::device_vector<null_order> d_null_precedence(null_precedence.begin(), null_precedence.end());
+
+  if (has_nulls(t)) {
+    auto ineq_op = (find_first)
+      ? row_lexicographic_comparator<true>(*d_t, *d_values, d_column_order.data().get(), d_null_precedence.data().get())
+      : row_lexicographic_comparator<true>(*d_values, *d_t, d_column_order.data().get(), d_null_precedence.data().get());
+
+    launch_search(count_it, count_it, t.num_rows(), values.num_rows(),
+                  result_view.data<size_type>(), ineq_op, find_first, stream);
+  } else {
+    auto ineq_op = (find_first)
+      ? row_lexicographic_comparator<false>(*d_t, *d_values, d_column_order.data().get(), d_null_precedence.data().get())
+      : row_lexicographic_comparator<false>(*d_values, *d_t, d_column_order.data().get(), d_null_precedence.data().get());
+
+    launch_search(count_it, count_it, t.num_rows(), values.num_rows(),
+                  result_view.data<size_type>(), ineq_op, find_first, stream);
+  }
+
+  return result;
+}
+
+template <typename Element, bool nullable = true>
+struct compare_with_value{
+  compare_with_value(column_device_view c, Element val, bool val_is_valid, bool nulls_are_equal)
+
+    : col{c}, value{val}, val_is_valid{val_is_valid}, nulls_are_equal{nulls_are_equal} {}
+
+  __device__ bool operator()(size_type i) noexcept {
+    if (nullable) {
+      bool const col_is_null{col.nullable() and col.is_null(i)};
+      if (col_is_null and not val_is_valid)
+        return nulls_are_equal;
+      else if (col_is_null == val_is_valid)
+        return false;
+    }
+    
+    return equality_compare<Element>(col.element<Element>(i), value);
+  }
+
+  column_device_view        col;
+  Element                   value;
+  bool val_is_valid;
+  bool nulls_are_equal;
+};
+
+template <typename Element>
+void populate_element(scalar const& value, Element &e) {
+  using ScalarType = cudf::experimental::scalar_type_t<Element>;
+  auto s1 = static_cast<const ScalarType *>(&value);
+
+  e = s1->value();
+}
+
+template <>
+void populate_element<string_view>(scalar const& value, string_view &e) {
+  using ScalarType = cudf::experimental::scalar_type_t<string_view>;
+  auto s1 = static_cast<const ScalarType *>(&value);
+
+  e = string_view{s1->data(), s1->size()};
+}
+  
+struct contains_scalar {
+  template <typename Element>
+  bool operator()(column_view const& col, scalar const& value,
+                  cudaStream_t stream,
+                  rmm::mr::device_memory_resource *mr) {
+
+    auto d_col = column_device_view::create(col, stream);
+    auto data_it = thrust::make_counting_iterator<size_type>(0);
+
+    bool    element_is_valid{value.is_valid()};
+    Element element;
+
+    populate_element(value, element);
```

Comment:
This should just use a `scalar_device_view`.

Reply:
I started that way and it became complicated, although I now know more about how strings and scalars work.  Will revisit in a new PR.